### PR TITLE
add raw createIndex function with upstream types

### PR DESCRIPTION
**Pull request by bmschmidt**
The types on createIndex are pretty vague and have some custom wrappings; adding a new function at createIndexRaw lets us use the API types directly without breaking back-compatibility.

----

> [!IMPORTANT]
> Adds `createIndexRaw` function in `AtlasDataset` for direct API type usage in index creation, with tests and dependency updates.
> 
>   - **Behavior**:
>     - Adds `createIndexRaw` function in `AtlasDataset` class in `src/project.ts` for direct API type usage in index creation.
>     - Uses `createIndexTextDefaults` for default values, allowing user options to override them.
>     - Ensures `project_id` is set in the request.
>   - **Tests**:
>     - Adds test for `createIndexRaw` in `project.test.ts` to verify index creation.
>   - **Dependencies**:
>     - Updates `apache-arrow` to `^13.0.0` in `package.json`.
>     - Removes `@types/jasmine` from `devDependencies` in `package.json`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=nomic-ai%2Fts-nomic&utm_source=github&utm_medium=referral)<sup> for 04f9d571312094c93c3aead88c5f360582a3009b. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by bmschmidt** [comment]
* **#82** <a href="https://app.graphite.dev/github/pr/nomic-ai/ts-nomic/82?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bmschmidt and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: tests/project.test.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { AtlasProjection } from '../src/projection';
 import { AtlasUser } from '../src/user';
 import { AtlasViewer } from '../src/viewer';
 import { AtlasOrganization } from '../src/organization';
+import { get } from 'http';
```
Comment (RLesser) [inline tests/project.test.ts:9]: remove?

**Review by RLesser — APPROVED** [review]
lgtm

**Comment by bmschmidt** [comment]
### Merge activity

* **Nov 4, 10:02 AM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/nomic-ai/ts-nomic/82).

**Pull request merged**